### ISSUE TITLE
Do not make GHCi output read-only

### ISSUE THREAD

**Issue by vshabanov** [haskell/haskell-mode#1437]
In old haskell-mode I used to be able to clean GHCi output (completely, or just few lines). But when I'm trying to select whole buffer (or line) and remove it I get `Text is read-only`.

Sometimes there are really long lines of Haskell data (10-100k characters for example) in GHCi output. And they slowdowns Emacs a lot. Plus sometimes I'd like to see only latest "fixed" data and do not mess with previous output.

Is it possible to make GHCi output editable again?

**Comment by ivan-m** [comment]
I think one thing that will make this difficult is the existence of the `*haskell-process-log*` buffer which will also contain all of this text.

My current solution is to kill the process, kill the log buffer and start it again, but then you lose the history :/

**Comment by vshabanov** [comment]
It's my current solution too. But it takes a lot of additional actions and time to reload everything.

**Comment by fice-t** [comment]
It's possible to make the read-only property optional.

Does `haskell-interactive-mode-clear` (`C-c C-k`) not work for you here? That function also clears out the process log.

**Comment by vshabanov** [comment]
Thank you for the `C-c C-k` hint! 

But I still would like to get more precise control. Sometimes I just want to select and remove a part of buffer: bad output, bad input with type errors, wrong input, too long output that obstructs the previous output.

And sometimes I'd like to copy a line using `C-k`. It actually works, line got copied, but it doesn't removed from buffer and there is a "Text is read only" warning, so it looks like it doesn't work.

In short I got used to old haskell-mode where output was easily editable and I could do anything with it.

**Comment by fice-t** [comment]
Alright. How fine-grained do you want the option to be? The easiest would be just a total `on/off` switch which all output (result, prompt, etc.) is either `read-only` or not. Or do you always want the prompt to be read-only?

> there is a "Text is read only" warning, so it looks like it doesn't work.

Changing the variable `kill-read-only-ok` could also help.

**Comment by vshabanov** [comment]
Thank you again for `kill-read-only-ok`.

Just the all output. So I could edit it any way I want. If prompts will be still read-only I'm afraid there will be problems with deleting text containing prompts.

**Comment by fice-t** [comment]
#1444 makes most output optional. I might have been too cautious and left a couple functions output read-only text since they were documented as `read-only`. Though if it's not enough for you then those should be easily changeable.

**Comment by vshabanov** [comment]
I think it should be totally read-only or totally read-write. Now when I set `haskell-interactive-mode-read-only` to `nil` I still can't select and remove several lines of output since prompts are still read-only. When I also set `haskell-interactive-prompt-read-only` to `nil` it behaves exactly like I want.

**Comment by fice-t** [comment]
I put that distinction in to allow for more fine-tuning of the property. Though with that implementation, if you set `haskell-interactive-mode-read-only` before loading `haskell-mode`, then the prompt will be the same read-only state as most of the other output.

That is, the default value of `haskell-interactive-prompt-read-only` is the value of `haskell-interactive-mode-read-only` when that file is loaded.

Though what's probably going to happen is that `haskell-interactive-mode-read-only` will be removed and you just have to set the prompt variable.

**Comment by Icelandjack** [comment]
I would like to run `json-mode-pretty-print-dwim` on GHCi output.
